src/notation/view/widgets/transposedialog.cpp

```diff
@@ -52,7 +52,10 @@ TransposeDialog::TransposeDialog(QWidget* parent)
     bool rangeSelection = selection()->isRange();
     setEnableTransposeKeys(rangeSelection);
     setEnableTransposeToKey(rangeSelection);
-    setEnableTransposeChordNames(rangeSelection);
+    bool hasChordNames = std::any_of(selection()->elements().begin(), selection()->elements().end(), [](EngravingItem* item) {
+        return item->isHarmony();
+    });
+    setEnableTransposeChordNames(hasChordNames);
```

Comment:
Couple of details:
- I'd create a local var for `selection()->elements()`
- hasChordNames -> hasChordSymbols
- could use const iterators
```suggestion
    bool rangeSelection = selection()->isRange();
    const std::vector<EngravingItem*>& elements = selection()->elements();
    setEnableTransposeKeys(rangeSelection);
    setEnableTransposeToKey(rangeSelection);
    bool hasChordSymbols = std::any_of(elements.cbegin(), elements.cend(), [](EngravingItem* item) {
        return item->isHarmony();
    });
    setEnableTransposeChordNames(hasChordSymbols);
```

Reply:
This is code that I assume will be replaced soon enough with a qml-based dialog. As it is, it would have been quite adequate just to have checkbox always checked/enabled, it doesn't do any harm.  And I deliberately called it "hasChordNames" to make the purpose clear (there are chord names present because there "harmony" elements exist in the selection).

Reply:
(BTW, what do you mean by const iterators? Would I would like to do is use an iterator that points to a const Engraving* item, but I'm not entirely sure how to do that...)

Reply:
> if I use cbegin()/cend(), surely it should be `[](const Engraving* item)`?

No; it would rather be `[](const (EngravingItem*)& item)` instead of `[]((EngravingItem*)& item)` (not sure if parentheses in type names are even valid, but this is just for illustration). I.e. since it's a list of pointers, the `const`-ness would apply to the pointers themselves and not to their pointees.

But by not writing the type as a reference but as a copy, which is fine for pointers, it's just a copy and we don't need to worry about whether the reference was const or not.

Still, using `[](const Engraving* item)` is possible anyway (even with non-const iterators) and makes sense in this case, to make clear that the lambda won't modify `item`.

And yes, such changes won't require re-testing.

My point about `chordNames` was that the synonym `chordSymbols` would make more sense because we use that terminology everywhere (except where we use "harmony").